lib/detect-testing-library-utils.ts

```diff
@@ -13,48 +18,84 @@ export function detectTestingLibraryUtils<
+        settings: TestingLibrarySettings;
+      }
+    >,
     optionsWithDefault: Readonly<TOptions>
-  ): TRuleListener => {
-    let isImportingTestingLibrary = false;
+  ): TSESLint.RuleListener => {
+    let isImportingTestingLibraryModule = false;
+    let isImportingCustomModule = false;
 
-    // TODO: init here options based on shared ESLint config
+    // Init options based on shared ESLint settings
+    const customModule = context.settings['testing-library/module'];
 
-    // helpers for Testing Library detection
+    // Helpers for Testing Library detection.
     const helpers: DetectionHelpers = {
-      getIsImportingTestingLibrary() {
-        return isImportingTestingLibrary;
+      getIsTestingLibraryImported() {
+        if (!customModule) {
+          return true;
+        }
+
+        return isImportingTestingLibraryModule || isImportingCustomModule;
+      },
+      canReportErrors() {
+        return this.getIsTestingLibraryImported();
       },
     };
 
-    // instructions for Testing Library detection
+    // Instructions for Testing Library detection.
+    // `ImportDeclaration` must be first in order to know if Testing Library
+    // is imported ASAP.
     const detectionInstructions: TSESLint.RuleListener = {
       ImportDeclaration(node: TSESTree.ImportDeclaration) {
-        isImportingTestingLibrary = /testing-library/g.test(
-          node.source.value as string
-        );
+        if (!isImportingTestingLibraryModule) {
+          // check only if testing library import not found yet so we avoid
+          // to override isImportingTestingLibraryModule after it's found
+          isImportingTestingLibraryModule = /testing-library/g.test(
+            node.source.value as string
+          );
+        }
+
+        if (!isImportingCustomModule) {
+          // check only if custom module import not found yet so we avoid
+          // to override isImportingCustomModule after it's found
+          const importName = String(node.source.value);
+          isImportingCustomModule = importName.endsWith(customModule);
+        }
       },
     };
 
     // update given rule to inject Testing Library detection
     const ruleInstructions = ruleCreate(context, optionsWithDefault, helpers);
-    const enhancedRuleInstructions = Object.assign({}, ruleInstructions);
+    const enhancedRuleInstructions: TSESLint.RuleListener = {};
+
+    // The order here is important too: detection instructions must come before
+    // than rule instructions to:
+    //  - detect Testing Library things before the rule is applied
+    //  - be able to prevent the rule about to be applied if necessary
+    const allKeys = new Set(
+      Object.keys(detectionInstructions).concat(Object.keys(ruleInstructions))
+    );
 
-    Object.keys(detectionInstructions).forEach((instruction) => {
-      (enhancedRuleInstructions as TSESLint.RuleListener)[instruction] = (
-        node
-      ) => {
+    allKeys.forEach((instruction) => {
```

Comment:
Now I iterate through **all** keys to be able to prevent original rule instructions.